### PR TITLE
JIT: revise checking for computed preds

### LINKED ISSUE #80193: JIT: build pred lists early, persist them throughout compilation
Currently there's a lag between incorporation of profile data (which happens quite early and uses its own flow edge representation) and building pred lists, and pred lists are sometimes dropped and recomputed from scratch (#49030).

We should be building pred lists early on and using those for profile incorporation, and then keep the lists valid and persistent throughout compilation.

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Since pred lists now exist pervasively, change most methods that had conditional pred list updates to assert preds exist and always update.

To make sure I got all uses, I renamed `fgComputePredsDone` to `fgPredsComputed`.

Remove the ability to drop EH, as it doesn't update pred lists properly and so has been broken for quite a while now.

Remove some of the logic for fixing up finally targets, since we now always have pred lists around and so don't need the blanket invaliation.

Closes #80193.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@EgorBo PTAL
cc @dotnet/jit-contrib 

This is the final bit of revision for pred lists. I may still try and encapsulate the remove/add ref patterns that are now fairly pervasive but will likely instead start building upon the new capabilities that persistent pred lists offer.

File: src/coreclr/jit/compiler.h [hunk 1]
```diff
@@ -4512,7 +4508,7 @@ class Compiler
 #endif // FEATURE_JIT_METHOD_PERF
 
     bool fgModified;             // True if the flow graph has been modified recently
-    bool fgComputePredsDone;     // Have we computed the bbPreds list
+    bool fgPredsComputed;        // Have we computed the bbPreds list
```
Comment (EgorBo) [inline src/coreclr/jit/compiler.h:4511]: Can you educate me why we need this at all? I assume it's `false` only during importing?

**Review by EgorBo — APPROVED** [review]

File: src/coreclr/jit/compiler.h (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/compiler.h:4511]: We actually build pred lists before importation, so `fgPredsComputed` is true for all phases. So it seems like something we don't need. But I kept this concept around for two reasons:
* we sometimes do a bit of flow graph modification very early before we have added preds lists, and call methods that are also called after we have pred edges; those methods need to behave differently before and after (mainly `fgEnsureFirstBBisScratch` and friends
* we want to make sure no other pred-list sensitive method is called early.

**Comment by AndyAyersMS** [comment]
Failure is known issue.

**Pull request merged**